e2e/cypress/integration/messaging/reactions_spec.js

```diff
@@ -0,0 +1,183 @@
+            cy.get('.emoji-picker__items #emoji-1f631').wait(TIMEOUTS.HALF_SEC).click();
+
+            // # Click the `+` button next to the existing reactions (visible on hover)
+            cy.get(`#addReaction-${postId}`).should('exist').click({force: true});
+
+            // # Click to select an emoji from the picker
+            cy.get('.emoji-picker__items #emoji-1f643').wait(TIMEOUTS.HALF_SEC).click();
+
+            // * Emoji reaction is added to the post
+            cy.get(`#${postId}_message`).within(() => {
+                cy.findByLabelText('reactions').should('exist');
+                cy.findByLabelText('remove reaction upside down face').should('exist');
+            });
+
+            // * Reaction appears in recently used section of emoji picker
+            cy.get('#emojiPickerButton').click().then(() => {
+                cy.findAllByTestId('emojiItem').first().within(($el) => {
+                    cy.wrap($el).findByTestId('upside_down_face').should('be.visible');
+                });
+            });
+        });
+    });
+
+    it('MM-T2192 RHS (reply) shows emoji picker for reactions - Reply box and plus icon', () => {
+        // # Post a message
+        cy.postMessage('Hello to you all');
+
+        cy.getLastPostId().then((postId) => {
+            // # Click a reply arrow to open reply RHS
+            cy.clickPostCommentIcon(postId).then(() => {
+                // # Click the expand arrows in top right to expand RHS
+                cy.findByLabelText('Expand').click();
+
+                // # Hover over the message, observe emoji picker icon
+                cy.get(`#RHS_ROOT_reaction_${postId}`).should('exist').click({force: true});
+
+                // # Click the emoji picker icon to react to the message, select a reaction
+                cy.get('.emoji-picker__items #emoji-1f631').wait(TIMEOUTS.HALF_SEC).click();
+
+                // # Hover over the post again and observe the `+` icon next to your reaction
+                cy.get(`#addReaction-${postId}`).should('exist').click({force: true});
+
+                // # Click the `+` icon and select a different reaction
+                cy.get('.emoji-picker__items #emoji-1f643').wait(TIMEOUTS.HALF_SEC).click();
+
+                // * Two reactions are added to the message in the expanded RHS
+                cy.get(`#rhsPost_${postId}`).within(() => {
+                    cy.findByLabelText('reactions').should('be.visible');
+                    cy.findByLabelText('remove reaction scream').should('be.visible');
+                    cy.findByLabelText('remove reaction upside down face').should('be.visible');
+                });
+
+                // # Close RHS
+                cy.closeRHS();
+            });
+        });
+    });
+
+    it('MM-T2195 Emoji reaction - not available on system message Flag - not available on system message Pin - not available on system message Can delete your own system message', () => {
```

Comment:
Hi @isacikgoz - please change all references of `flag` to `save` for this test.  (Flagged posts recently renamed to Saved posts).  Thanks!

Reply:
🤦